Nodejs/Product/Analysis/JavaScript/declaration.cs

```diff
@@ -56,7 +56,8 @@ public override IEnumerable<Node> Children
 
             set {
                 for (int i = 0; i < value.Length; i++) {
-                    value[i].Parent = this;
+                    if (value[i] != null)
```

Comment:
Hrm, I don't believe any of the values there should ever be null. Looks like `ParseIdentifierInitializer` is returning null, and that value is not being null checked before being added to `varList` in `ParseForStatement`. Additionally, I don't believe we should be calling `UpdateWithOtherNode` unless `initializer` is not null. So instead of checking for null within , I think we should just do the proper null checks within `ParseForStatement` before adding it to the list.
* https://github.com/Microsoft/nodejstools/blob/master/Nodejs/Product/Analysis/JavaScript/jsparser.cs#L1111-L1112
* https://github.com/Microsoft/nodejstools/blob/master/Nodejs/Product/Analysis/JavaScript/jsparser.cs#L1119-L1120

@DinoV could you please confirm?